Improvement/login/changes to login screen

mariagiorgetti opened the pull request.

DaikoneKisu committed:
fix: resolve a conflict in tailwind.config.js when merge with feature/login branch

mariagiorgetti committed:
proyecto fit, tarea 1

DaikoneKisu commented on package-lock.json:
¿Estás usando npm? Debes usar pnpm y borrar este archivo para evitar fallos de compatibilidad

Hector1567XD replied:
De pana

DaikoneKisu commented on src/lib/components/sign-up-log-in-switch-button/SignUpLogInSwitchButton.svelte:
@Noralgorithm ¿Debiera seguir existiendo este archivo? Como ya no tenemos sign up, lo veo innecesario, más que por retrocompatibilidad

Noralgorithm replied:
No debería

DaikoneKisu commented on src/lib/components/sign-up-log-in-switch-button/sign-up-log-in-switch-button.test.ts:
@Noralgorithm lo mismo que para el archivo que testea

Noralgorithm replied:
Tampoco debería

DaikoneKisu commented on src/lib/images/edificio.png:
¿Quizás un cambio de nombre de edificio.png a login-image.png le caería bien? @Noralgorithm @Paulaforero

Noralgorithm replied:
Sí sería un mejor nombre, igual el nombre puede que cambie a futuro si se llega a implementar lo de las múltiples imágenes cambiantes.

DaikoneKisu commented on src/routes/(auth)/sign-up/+page.svelte:
Borrar este archivo y todo /sign-up

Noralgorithm replied:
Sí, se deben borrar porque no tenemos registro en nuestra plataforma por los momentos

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+<script lang="ts">
+	import './layout.css';
```
¿Por qué usar css si usamos tailwind? Borrar esta línea en cuanto se adapten los estilos que están ahí a tailwind

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+<script lang="ts">
+	import './layout.css';
+	import '../../app.css'
```
Dejar de importar esto aquí

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+	import '../../app.css'
+	import { page } from '$app/stores'
+	import ucabLogo from '$lib/images/logo-ucab.webp'
+	import firstLoginImage from '$lib/images/edificio.png'
```
```suggestion
	import loginImage from '$lib/images/login-image.png'
```
Hacer eso si se decide cambiarle el nombre a login-image.png, sino:
```suggestion
	import edificio from '$lib/images/edificio.png'
```

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+	import { page } from '$app/stores'
+	import ucabLogo from '$lib/images/logo-ucab.webp'
+	import firstLoginImage from '$lib/images/edificio.png'
+	import secondLoginImage from '$lib/images/login-image-two.png'
```
```suggestion
```
Borrar esto porque no se usa

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+	import ucabLogo from '$lib/images/logo-ucab.webp'
+	import firstLoginImage from '$lib/images/edificio.png'
+	import secondLoginImage from '$lib/images/login-image-two.png'
+	import SignUpLogInSwitchButton from '$lib/components/sign-up-log-in-switch-button/SignUpLogInSwitchButton.svelte'
```
```suggestion
```
Borrar esto porque no se usa

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+	import firstLoginImage from '$lib/images/edificio.png'
+	import secondLoginImage from '$lib/images/login-image-two.png'
+	import SignUpLogInSwitchButton from '$lib/components/sign-up-log-in-switch-button/SignUpLogInSwitchButton.svelte'
+
+	let chosen: 'Iniciar sesión' | 'Registrarse' = 'Iniciar sesión'
+
+	$: if ($page.url.pathname === '/sign-up') {
+		chosen = 'Registrarse'
+	} else if ($page.url.pathname === '/login') {
+		chosen = 'Iniciar sesión'
+	}
```
```suggestion
```
Borrar todo esto porque no se usa

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+
+<main class="flex">
+    <div class="min-h-screen flex flex-col items-center justify-start  pb-24 pt-10 w-1/2">
+        <img src={ucabLogo} alt="logo de la ucab" class="max-w-[497px] aspect-[497/75] overflow-hidden mt-custom" />
```
```suggestion
        <img src={ucabLogo} alt="logo de la ucab" class="max-w-[497px] aspect-[497/75] overflow-hidden mt-16" />
```
Usar verdaderas clases de tailwind, no inventes las tuyas propias

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+
+
+<main class="flex">
+    <div class="min-h-screen flex flex-col items-center justify-start  pb-24 pt-10 w-1/2">
```
¿Por qué este div? ¿Por qué no aplicar estas clases al elemento main? Intentar refactorizar para que sea posible

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+        </div>
+    </div>
+    <div>
+        <img src={firstLoginImage} alt=""
```
```suggestion
        <img src={loginImage} alt=""
```
o
```suggestion
        <img src={edificio} alt=""
```
dependiendo de cuál nombre se acepte para ese archivo

DaikoneKisu commented on src/routes/(auth)/+layout.svelte:
```diff
@@ -0,0 +1,49 @@
+    </div>
+    <div>
+        <img src={firstLoginImage} alt=""
+            style="
+				margin-top: 20px;
+                width: 595px;
+                height: 673px;
+                position: absolute;
+                top: 29px6
+                left: 720px;
+                border-radius: 10px 190px 10px 190px;
+            "
```
```suggestion
```
Usar tailwind para esto

DaikoneKisu replied:
Aparte, ¿por qué posicionar la imagen absolutamente?

DaikoneKisu replied:
Incluso, ¿por qué posicionar la imagen dentro del elemento main? Debiera estar fuera

DaikoneKisu commented on src/routes/(auth)/layout.css:
Pasar los estilos de este archivo a +layout.svelte en forma de clases de tailwind y borrar este archivo

DaikoneKisu commented on src/routes/(authed)/+page.svelte:
```diff
@@ -0,0 +1,8 @@
+<script lang="ts">
+	import '../../app.css'
```
```suggestion
```
Borrarlo porque debiera de importarse en el +layout.svelte en /src/routes

DaikoneKisu commented on src/routes/+layout.svelte:
```suggestion
<script>
	import '../app.css'
</script>

<slot />
```
Dicho de otro modo, volver a crear este archivo con sus contenidos previos

DaikoneKisu commented on src/lib/images/login-image-one.png:
Borrar esta imagen porque no se usa (creo, chequear bien)

Noralgorithm replied:
Confirmo

DaikoneKisu commented on src/lib/images/login-image-two.png:
Borrar esta imagen porque no se usa

DaikoneKisu reviewed (changes requested):
Revisar

Noralgorithm commented on src/lib/components/input/Input.svelte:
Hay que mergear la rama feature/input para tener los últimos cambios presentes acá.

Paulaforero reviewed (changes requested):
Las proporciones de toda la página son muy grades, recomiendo hacer los elementos más pequeños en la página tienes el tamaño de los inputs en 400x76 y en el figma estan en 330x64. Para el tamaño del bienvenidos y el logo de la universidad trata de que el bienvenidos este alineado con el logo que esten del mismo tamaño, a penas comienza el logo abajo comienza el bienvenidos, se soluciona haciendo el bienvenidos más pequeño (ya mando una foto 
![photo1703177902](https://github.com/DaikoneKisu/uchamba-frontend/assets/94322466/a0c65042-62f4-4809-b5d8-dd37f3a3f0f4)
). La foto esta bien pero recuerda que los bordes cambiaron y ya no son así (adjunto otra imágen 
![photo1703178117](https://github.com/DaikoneKisu/uchamba-frontend/assets/94322466/6af40076-42dd-47a6-8a1d-4cc2294ca77a)
). Para el diseño responsive a pesar de que no esta en el figma (falta de tiempo) lo que harás es que cuando la página se haga más pequeña se eliminará la foto y el form de iniciar sesión quedará en el medio (cuando digo el form de iniciar sesión me refiero a esto 
![photo1703178311](https://github.com/DaikoneKisu/uchamba-frontend/assets/94322466/29ca2403-116d-4416-b659-aa95fd970bc9)
) Cualquier duda o algo que no hayas entendido me avisas y vamos al dc

Paulaforero commented:
Para los mensajes de error después de que se escribe algo en los inputs hay que alinearlos a la izquierda y que no se rueda para abajo los inputs

mariagiorgetti committed:
cambiosInicioSesion

DaikoneKisu committed:
Merge branch 'feature/prettier' into feature/figma-text-styles

DaikoneKisu committed:
feat: tell tailwind to apply dark mode depending on class prefix

DaikoneKisu committed:
style: fix app.css styles

Paulaforero reviewed (changes requested):
La imagen esta bien, pero te falta poner los títulos un poco más pequeños, además los mensajes de error de los inputs tienen que estar alineados a la izquierda, no deben estar centrados, y no se deben de rodar los inputs cuando aparece esos mensajes. 
![Uploading photo1703630467.jpeg…]()

Las letras del botón deben de estar en blanco 
![photo1703630205](https://github.com/DaikoneKisu/uchamba-frontend/assets/94322466/f84f0e9c-331f-4cda-b52c-0e184508c85a)

DaikoneKisu committed:
Merge branch 'staging' into feature/figma-text-styles to resolve conflicts

Noralgorithm committed:
refactor: move type to lib folder

Noralgorithm committed:
style: line between imports

Paulaforero committed:
feat: change space in inputs in form

DaikoneKisu committed:
chore: update dependencies

DaikoneKisu committed:
feat!: delete sign-up

DaikoneKisu committed:
feat!: change login image file

BREAKING CHANGE: delete previous login image

DaikoneKisu committed:
feat: add props, error displaying and change markup/style

DaikoneKisu committed:
fix: add missing once modifier

DaikoneKisu committed:
Merge branch 'feature/input-with-icon' into improvement/login/changes-to-login-screen

DaikoneKisu committed:
feat: change input label size to 0.9375rem and input text size to default

DaikoneKisu committed:
Merge branch 'feature/input-with-icon' into improvement/login/changes-to-login-screen

DaikoneKisu committed:
feat: add animation/functionality and change markup/styles

DaikoneKisu committed:
feat: add autocomplete prop and change error font-size

DaikoneKisu committed:
feat: change error font-size

DaikoneKisu committed:
Merge branch 'feature/input-with-icon' into improvement/login/changes-to-login-screen

DaikoneKisu committed:
Merge branch 'feature/password-input' into improvement/login/changes-to-login-screen

DaikoneKisu committed:
feat!: delete sign-up related stuff

DaikoneKisu committed:
feat!: improve login screen
>>
>> change styles and props in a lot of stuff
>> change markup to not have so many divs
>> delete onMount callback
>> delete enabling of submit form button on form.reason
>> change from previous error showing impl to new one via input's own error props
>> change user input validation to onchange instead of oninput
>>
>> BREAKING CHANGE: delete layout.css in auth

DaikoneKisu committed:
feat: add strong styles and fix import warnings

DaikoneKisu committed:
fix: properly import fonts used

DaikoneKisu committed:
feat: add topmost container with padding

DaikoneKisu committed:
refactor: add lang=ts to script in routes layout

DaikoneKisu committed:
feat!: delete surrounding div in +layout.svelte

DaikoneKisu committed:
feat!: change font-sizes to rems and add h3 styles

DaikoneKisu committed:
Merge branch 'feature/figma-text-styles' into improvement/login/changes-to-login-screen

DaikoneKisu committed:
chore: update dependencies lock

DaikoneKisu committed:
fix!: delete emptying style

DaikoneKisu committed:
fix: delete duplicate function impl

DaikoneKisu committed:
fix: close form element

DaikoneKisu committed:
feat: add padding and min height to login

DaikoneKisu committed:
feat!: not even I know what happened
